projects/plugins/jetpack/extensions/blocks/anchor-fm/anchor-fm.php

```diff
@@ -105,10 +105,18 @@ function process_anchor_params() {
 		if ( ! \is_wp_error( $rss ) ) {
 			update_post_meta( $post->ID, 'jetpack_anchor_podcast', $podcast_id );
 
+			// If we haven't got an episode ID, try and get the latest episode.
+			if ( empty( $episode_id ) && $rss->get_item_quantity() ) {
+				$latest_episode = $rss->get_item( 0 );
+				if ( $latest_episode ) {
+					$episode_id = $latest_episode->get_id();
+				}
+			}
+
 			if ( ! empty( $episode_id ) ) {
 				$track = $podcast_helper->get_track_data( $episode_id );
 				if ( ! \is_wp_error( $track ) ) {
-					update_post_meta( $post->ID, 'jetpack_anchor_episode', $episode_id );
+					update_post_meta( $post->ID, 'jetpack_anchor_episode', $track['guid'] );
```

Comment:
Any specific reason you swapped `$track['guid']` to `$episode_id`?

(Intuitively, it makes sense... store the thing you just used to look up an episode for when you need to do the same lookup in the future. I'm just curious if you found this made a difference in some way.)

Reply:
No, it came from when I was using a default parameter with `get_track_data`, but as you say, it struck me as more intuitive.